util/dvsim/FpvCfg.py

```diff
@@ -0,0 +1,213 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import csv
+import os
+import logging as log
+from pathlib import Path
+
+from tabulate import tabulate
+
+from OneShotCfg import OneShotCfg
+from utils import subst_wildcards
+
+
+class FpvCfg(OneShotCfg):
+    """Derivative class for fpv purposes.
+    """
+    def __init__(self, flow_cfg_file, proj_root, args):
+        super().__init__(flow_cfg_file, proj_root, args)
+        self.header = ["name", "proven", "cex", "undetermined", "covered", "unreachable",
+                       "pass_rate", "cov_rate"]
+        self.summary_header = ["name", "pass_rate", "stimuli_cov", "coi_cov", "prove_cov"]
+
+    def __post_init__(self):
+        super().__post_init__()
+        self.results_title = self.name.upper() + " FPV Results"
+
+    def gen_results_summary(self):
+        # '''
+        # Gathers the aggregated results from all sub configs
+        # The result summary will only contains a percentage to the passing
+        # rate, Stimuli coverage, COI coverage, and Proven coverage
+        # '''
+        log.info("Create summary of FPV results")
+
+        results_str = "## " + self.results_title + " (Summary)\n\n"
+        results_str += "### " + self.timestamp_long + "\n\n"
+
+        colalign = ("center", ) * len(self.summary_header)
+        table = [self.summary_header]
+        for cfg in self.cfgs:
+            for key in cfg.result_summary:
+                try:
+                    table.append(cfg.result_summary[key])
+                except Exception as e:
+                    table.append([key, "ERROR", "N/A", "N/A", "N/A"])
+                    log.error("cfg:%s item: %s could not find generated result: %s", cfg, key, e)
+        if len(table) > 1:
+            self.results_summary_md = results_str + tabulate(
+                table, headers="firstrow", tablefmt="pipe", colalign=colalign)
+        else:
+            self.results_summary_md = results_str
+
+        print(self.results_summary_md)
+
+        # Generate email results summary
+        colalign = ("left", ) + ("center", ) * (len(self.header) - 1)
+        email_table = [self.header]
+
+        for cfg in self.cfgs:
+            email_table.append([
+                cfg.name,
+                str(cfg.result["proven"]) + " G ",
+                str(cfg.result["cex"]) + " E ",
+                str(cfg.result["undetermined"]) + " W ",
+                str(cfg.result["covered"]) + " G ",
+                str(cfg.result["unreachable"]) + " W ",
+                cfg.result["pass_rate"],
+                cfg.result["cov_rate"]])
+        if len(email_table) > 1:
+            self.email_summary_md = results_str + tabulate(
+                email_table, headers="firstrow", tablefmt="pipe", colalign=colalign)
+
+        return self.results_summary_md
+
+    def _gen_results(self):
```

Comment:
I think it would be helpful if we aggregated and presented tool warnings and errors here as well. that way we will know whether an FPV pass completed correctly or not. e.g., sometimes runs are not successful because of syntax errors or similar... 

I think you already have an `["errors"]` field in the parsed out hjson data structure that is populated if the logfiles cannot be read. we could copy over the regex search mechanism from lint and adapt the patterns for FPV to cover more tool warnings and errors:
https://github.com/lowRISC/opentitan/blob/b33ffdaa35c399080d95042fd79789d276e4384b/hw/lint/tools/ascentlint/parse-lint-report.py#L40-L62

we just have to be careful with the messages themselves, since we may not be able to print them out publicly. but reporting the error/warning count (as e.g. done for synthesis) should be ok.